Fix flaky test: NPE in `Scalar.value()` returning null

Flaky test fails with NPE when `Scalar.value()` returns null and `toString()` is called on the result.

From this [run](https://github.com/objectionary/eo/actions/runs/14493036498/job/40653738703?pr=4097):

```java
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.toString()" because the return value of "org.cactoos.Scalar.value()" is null
      at org.cactoos.text.TextOfScalar.asString (TextOfScalar.java:39)
      at org.cactoos.text.TextEnvelope.asString (TextEnvelope.java:32)
      at org.cactoos.text.TextEnvelope.asString (TextEnvelope.java:32)
      at org.cactoos.text.TextOfScalar.asString (TextOfScalar.java:39)
      at org.cactoos.text.TextEnvelope.asString (TextEnvelope.java:32)
      at org.cactoos.text.Split.lambda$new$0 (Split.java:108)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.cactoos.iterable.IterableOf.iterator (IterableOf.java:64)
      at org.cactoos.iterable.Mapped.lambda$new$0 (Mapped.java:42)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.cactoos.iterable.IterableOf.iterator (IterableOf.java:64)
      at org.cactoos.iterable.IterableEnvelope.iterator (IterableEnvelope.java:34)
      at org.cactoos.iterable.IterableEnvelope.iterator (IterableEnvelope.java:34)
      at org.cactoos.iterable.Filtered.lambda$new$0 (Filtered.java:52)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.cactoos.iterable.IterableOf.iterator (IterableOf.java:64)
      at org.cactoos.iterable.IterableEnvelope.iterator (IterableEnvelope.java:34)
      at org.cactoos.iterable.Filtered.lambda$new$0 (Filtered.java:52)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.cactoos.iterable.IterableOf.iterator (IterableOf.java:64)
      at org.cactoos.iterable.IterableEnvelope.iterator (IterableEnvelope.java:34)
      at org.cactoos.scalar.ItemAt.lambda$new$1 (ItemAt.java:90)
      at org.cactoos.scalar.Sticky.lambda$new$0 (Sticky.java:54)
      at org.cactoos.func.StickyFunc.lambda$new$0 (StickyFunc.java:51)
      at org.cactoos.func.StickyBiFunc.apply (StickyBiFunc.java:74)
      at org.cactoos.func.StickyFunc.apply (StickyFunc.java:58)
      at org.cactoos.scalar.Sticky.value (Sticky.java:60)
      at org.cactoos.scalar.ItemAt.value (ItemAt.java:108)
      at org.cactoos.scalar.FirstOf.value (FirstOf.java:85)
      at org.cactoos.scalar.Mapped.lambda$new$0 (Mapped.java:39)
      at org.cactoos.scalar.ScalarEnvelope.value (ScalarEnvelope.java:34)
      at org.cactoos.text.TextOfScalar.asString (TextOfScalar.java:39)
      at org.cactoos.text.TextEnvelope.asString (TextEnvelope.java:32)
      at org.cactoos.text.TextEnvelope.asString (TextEnvelope.java:32)
      at org.cactoos.text.Split.lambda$new$0 (Split.java:108)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.cactoos.iterable.IterableOf.iterator (IterableOf.java:64)
      at org.cactoos.iterable.Mapped.lambda$new$0 (Mapped.java:42)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.cactoos.iterable.IterableOf.iterator (IterableOf.java:64)
      at org.cactoos.iterable.IterableEnvelope.iterator (IterableEnvelope.java:34)
      at org.cactoos.iterable.IterableEnvelope.iterator (IterableEnvelope.java:34)
      at org.cactoos.iterable.Filtered.lambda$new$0 (Filtered.java:52)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.cactoos.iterable.IterableOf.iterator (IterableOf.java:64)
      at org.cactoos.iterable.IterableEnvelope.iterator (IterableEnvelope.java:34)
      at org.cactoos.scalar.ItemAt.lambda$new$1 (ItemAt.java:90)
      at org.cactoos.scalar.Sticky.lambda$new$0 (Sticky.java:54)
      at org.cactoos.func.StickyFunc.lambda$new$0 (StickyFunc.java:51)
      at org.cactoos.func.StickyBiFunc.apply (StickyBiFunc.java:74)
      at org.cactoos.func.StickyFunc.apply (StickyFunc.java:58)
      at org.cactoos.scalar.Sticky.value (Sticky.java:60)
      at org.cactoos.scalar.ItemAt.value (ItemAt.java:108)
      at org.cactoos.scalar.FirstOf.value (FirstOf.java:85)
      at org.cactoos.scalar.Mapped.lambda$new$0 (Mapped.java:39)
      at org.cactoos.scalar.ScalarEnvelope.value (ScalarEnvelope.java:34)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.eolang.maven.ChText.inside (ChText.java:108)
      at org.eolang.maven.ChText.value (ChText.java:63)
      at org.eolang.maven.ChRemote.value (ChRemote.java:39)
      at org.eolang.maven.ChNarrow.validHash (ChNarrow.java:40)
      at org.eolang.maven.ChNarrow.value (ChNarrow.java:30)
      at org.cactoos.scalar.Sticky.lambda$new$0 (Sticky.java:54)
      at org.cactoos.func.StickyFunc.lambda$new$0 (StickyFunc.java:51)
      at org.cactoos.func.StickyBiFunc.apply (StickyBiFunc.java:74)
      at org.cactoos.func.StickyFunc.apply (StickyFunc.java:58)
      at org.cactoos.scalar.Sticky.value (Sticky.java:60)
      at org.cactoos.scalar.Checked.value (Checked.java:56)
      at org.cactoos.scalar.IoChecked.value (IoChecked.java:44)
      at org.cactoos.scalar.Unchecked.value (Unchecked.java:37)
      at org.eolang.maven.ChCached.value (ChCached.java:33)
      at org.eolang.maven.MjUnphi.lambda$exec$1 (MjUnphi.java:99)
      at org.eolang.maven.Threaded.lambda$total$0 (Threaded.java:56)
      at org.cactoos.scalar.CallableEnvelope.call (CallableEnvelope.java:34)
      at java.util.concurrent.FutureTask.run (FutureTask.java:317)
      at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
      at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
      at java.lang.Thread.run (Thread.java:1583)
```

--- a/.gitignore
+++ b/.gitignore
@@ -15,3 +15,5 @@ eo-runtime/measures.csv
 node_modules/
 target/
 xs3p.xsl_*
+.aidy
+/.aidy/

--- a/eo-maven-plugin/src/main/java/org/eolang/maven/ChCached.java
+++ b/eo-maven-plugin/src/main/java/org/eolang/maven/ChCached.java
@@ -5,6 +5,7 @@
 package org.eolang.maven;
 
 import org.cactoos.scalar.Sticky;
+import org.cactoos.scalar.Synced;
 import org.cactoos.scalar.Unchecked;
 
 /**
@@ -25,7 +26,7 @@ final class ChCached implements CommitHash {
      * @param delegate Delegate
      */
     ChCached(final CommitHash delegate) {
-        this.delegate = new Unchecked<>(new Sticky<>(delegate::value));
+        this.delegate = new Unchecked<>(new Synced<>(new Sticky<>(delegate::value)));
     }
 
     @Override

--- a/eo-maven-plugin/src/main/java/org/eolang/maven/CommitHashesText.java
+++ b/eo-maven-plugin/src/main/java/org/eolang/maven/CommitHashesText.java
@@ -8,7 +8,9 @@ import com.jcabi.aspects.RetryOnFailure;
 import com.jcabi.log.Logger;
 import java.net.URL;
 import java.util.concurrent.TimeUnit;
+import org.cactoos.Text;
 import org.cactoos.text.Sticky;
+import org.cactoos.text.Synced;
 import org.cactoos.text.TextEnvelope;
 import org.cactoos.text.TextOf;
 
@@ -31,7 +33,15 @@ final class CommitHashesText extends TextEnvelope {
      * Constructor.
      */
     CommitHashesText() {
-        super(new Sticky(CommitHashesText::asText));
+        this(CommitHashesText::asText);
+    }
+
+    /**
+     * Constructor.
+     * @param source Text source.
+     */
+    private CommitHashesText(final Text source) {
+        super(new Synced(new Sticky(source)));
     }
 
     /**

--- a/eo-maven-plugin/src/test/java/org/eolang/maven/ChCachedTest.java
+++ b/eo-maven-plugin/src/test/java/org/eolang/maven/ChCachedTest.java
@@ -4,9 +4,11 @@
  */
 package org.eolang.maven;
 
+import com.yegor256.Together;
 import java.util.concurrent.atomic.AtomicInteger;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
+import org.junit.jupiter.api.RepeatedTest;
 import org.junit.jupiter.api.Test;
 
 /**
@@ -34,6 +36,31 @@ final class ChCachedTest {
         }
         MatcherAssert.assertThat(
             "The delegate should be called exactly once, but it was not",
+            invocations.get(),
+            Matchers.equalTo(1)
+        );
+    }
+
+    @RepeatedTest(10)
+    void cachesHashInConcurrentEnvironment() {
+        final AtomicInteger invocations = new AtomicInteger(0);
+        final String tag = "parallel";
+        final ChCached cached = new ChCached(
+            () -> {
+                invocations.incrementAndGet();
+                return tag;
+            }
+        );
+        MatcherAssert.assertThat(
+            "We expect that all values are equal to the tag",
+            new Together<>(30, i -> cached.value())
+                .asList()
+                .stream()
+                .allMatch(tag::equals),
+            Matchers.is(true)
+        );
+        MatcherAssert.assertThat(
+            "The delegate should be called exactly once in concurrent environment, but it was not",
             invocations.get(),
             Matchers.equalTo(1)
         );

--- a/eo-maven-plugin/src/test/java/org/eolang/maven/CommitHashesTextTest.java
+++ b/eo-maven-plugin/src/test/java/org/eolang/maven/CommitHashesTextTest.java
@@ -30,10 +30,11 @@ final class CommitHashesTextTest {
 
     @Test
     void isThreadSafe() {
+        final CommitHashesText text = new CommitHashesText();
         MatcherAssert.assertThat(
             "Can be used in different threads without NPE",
             new Together<>(
-                thread -> new CommitHashesText().asString() != null
+                thread -> text.asString() != null
             ),
             Matchers.not(Matchers.hasItems(false))
         );

